app/controllers/application_controller/ci_processing.rb

```diff
@@ -1344,6 +1344,10 @@ def get_error_message_from_fog(ex)
 
   private ############################
 
+  def record_ineligible_to_retire(vms)
+    VmOrTemplate.includes_template?(vms.map(&:to_i).uniq)
```

Comment:
@h-kataria can we move this logic to the model? To me it looks like `MiqTemplate` should not support retirement operation whereas other subclasses of `VmOrTemplate` might do so.

cc @jameswnl